### ISSUE TITLE
Carthage support

### ISSUE THREAD

**Issue by ale84** [cloudinary/cloudinary_ios#47]
Could you include support for Carthage?

**Comment by FiRiN59** [comment]
Any news on this one?

**Comment by taragano** [comment]
Carthage is still not supported a the moment.
It's on our road-map to support it in one of our future versions, however there's no ETA for this yet.

**Comment by MartinBergerDX** [comment]
It would be useful to have this enhancement done.

**Comment by MartinBergerDX** [comment]
@taragano Are there some issues with adding Carthage support? It shouldn't take long.
 Maybe i can try?

**Comment by SandeepAggarwal** [comment]
Hey guys , pls include support for Carthage or open source the objc version of Cloudinary SDK

**Comment by daveingle** [comment]
@SandeepAggarwal until this issue is resolved you can use my fork of the repo. I've made the build scheme shared and switched to Carthage for the Alamofire dependency.

https://github.com/daveingle/cloudinary_ios

**Comment by SandeepAggarwal** [comment]
@daveingle Have u tested it whether it is running fine via carthage ?
Actually I also tried but couldn't make it work

**Comment by daveingle** [comment]
@SandeepAggarwal Yes, it's working for me, remember you need to include the Alamofire sub-dependency. I have managed to use Cloudinary on both the simulator and the device. Admittedly I have not tried a release build.

**Comment by SandeepAggarwal** [comment]
@daveingle Thank you, this was much needed in my ongoing project.
You saved my lot of headache. Thanks man.

**Comment by SandeepAggarwal** [comment]
@daveingle  I am getting an error upon using the cloudinary  in Carthage:
_

> Parse error: expected submodule commit SHA in output of task (ls-tree -z 2.0.1 Cloudinary/Frameworks/Alamofire) but encountered:

_ 
The cartfile looks like:

```
github "Alamofire/Alamofire" ~> 4.1
github "daveingle/cloudinary_ios"
```

**Comment by daveingle** [comment]
@SandeepAggarwal Carthage will use the latest tag if you don't have version info I believe.
Try `github "daveingle/cloudinary_ios" "master"`

**Comment by joel-yc** [comment]
Carthage support is a must for my team.  We can't use cocoapods.  

+1 for adding carthage support please

**Comment by runmad** [comment]
Any updates on this? We cannot use Cocapods and I have little luck integrating it manually with iOS 11 + Xcode 9.

**Comment by joel-yc** [comment]
We ended up not using the Cloudinary lib at all.  Instead, since we were already using Alamofire, we just did a direct multiform post to Cloudinary.

Alamofire.upload(multipartFormData: { multipartFormData in
      for (key, value) in parameters { ... }

Hope that helps...

**Comment by SandeepAggarwal** [comment]
hey @joel-yc , @runmad , give a try here https://github.com/SandeepAggarwal/cloudinary_ios

Just add following to your Cartfile:
github "https://github.com/SandeepAggarwal/cloudinary_ios.git" "master"

**Comment by runmad** [comment]
@SandeepAggarwal doesn't work for me

**Comment by daveingle** [comment]
@runmad That was forked from my repo, I've managed to release to the store using Carthage. What doesn't work for you.

**Comment by lacyrhoades** [comment]
Not sure about @runmad but this is what I get. It's pretty much the same if I try to `carthage update` from `cloudinary/cloudinary_ios`

```
$ carthage update
*** Fetching cloudinary_ios
*** Checking out cloudinary_ios at "ae86625fa1e7ce83e0b32f70080aef9398e4ac6e"
Parse error: expected submodule commit SHA in output of task (ls-tree -z ae86625fa1e7ce83e0b32f70080aef9398e4ac6e Cloudinary/Frameworks/Alamofire) but encountered: 
```

**Comment by daveingle** [comment]
Try using `carthage update --platform iOS --use-ssh`

**Comment by gemedet** [comment]
+1

**Comment by irodrigo17** [comment]
+1, this would be useful shouldn't be that hard to do

**Comment by shirlymanor** [comment]
I've added a request to our product team to add it to our road-map. we will update with any insights.

**Comment by meandkareem** [comment]
+1

**Comment by davidvpe** [comment]
@shirlymanor how is this progressing?

**Comment by shirlymanor** [comment]
@daveingle unfortunately there is no ETA yet. I've asked to higher the priority of this issue.

**Comment by daveingle** [comment]
@shirlymanor I think you meant to tag @davidvpe 

FWIW, The amount of time this task should take is minimal. Hours not days.

Although I no longer use Cloudinary, if I was reviewing and comparing services right now, not offering support for one of the most common dependency managers in the community is a big red flag.

This issue is from 2016 and hasn't been resolved, I don't think your product team understands the impact of having a ticket like this sitting open.

**Comment by shirlymanor** [comment]
Thank you for your feedback, we escalating this issue now and will update here when we will hear back.

**Comment by tevghenii** [comment]
Hi
Any news on Carthage support?

**Comment by eyalktCloudinary** [comment]
@tevghenii Thank you for your interest in this. There aren't any news available yet, we'll post a notification in this thread on any updates available.

**Comment by daveingle** [comment]
I needed to integrate Cloudinary again. Since I still prefer Carthage over Cocoapods I created a new fork and made the changes needed to get it going. Ripping out the Alamofire submodule and reintegrating that too using Carthage etc.

For those who it would help, just add this to your Cartfile:
```
github "daveingle/cloudinary_ios" "carthage-support"
```

Love it if you could drop Alamofire as a nested dependency sometime.

**Comment by ronen-burin** [comment]
@daveingle, thank you. your suggestion will be evaluated.

**Comment by meandkareem** [comment]
Any updates on this one ?

**Comment by ChristopherJoonMiller** [comment]
I would just like to state that I am also interested in using Carthage over cocoapods. I am hoping there is an official support option implemented soon.

**Comment by shirlymanor** [comment]
Hey, we published a beta release that supports carthage: https://github.com/cloudinary/cloudinary_ios/releases/tag/2.9.0-beta.2
Can you please let us know any feedback?

**Comment by ChechkovEugene** [comment]
In Version 2.9.0 all is fine. But in version 2.10.0 there is an error on installing as Carthage dependency

Dependency "cloudinary_ios" has no shared framework schemes for any of the platforms: iOS

If you believe this to be an error, please file an issue with the maintainers at https://github.com/cloudinary/cloudinary_ios/issues/new

**Comment by michalkcloudinay** [comment]
Hi @ChechkovEugene, thanks for reporting this. We have opened an issue internally and we will update once this is fixed.

**Comment by michalkcloudinay** [comment]
Hi @ChechkovEugene, A new version for iOS SDK was released (2.10.1) with a fix for this issue. Please let us know if everything works as expected. Thanks
